### PR TITLE
Handle hardlink or regular file for /etc/localtime

### PR DESCRIPTION
**Pull request by yunzheng**
A symlink is not always guaranteed for /etc/localtime.
This adds extra handling for hardlinks and regular files.

Issue found via: https://github.com/fox-it/citrix-netscaler-triage/issues/4

### CODE REVIEW

File: dissect/target/plugins/os/unix/locale.py [hunk 1]
```diff
@@ -14,6 +14,15 @@
 )
 
 
+def timezone_from_path(path):
```
Comment (Schamper) [inline dissect/target/plugins/os/unix/locale.py:17]:
```suggestion
def timezone_from_path(path: Path) -> str:
```

File: tests/test_plugins_os_unix_locale.py [hunk 2]
```diff
@@ -29,3 +31,43 @@ def test_locale_plugin_unix_quotes(target_unix_users, fs_unix):
     target_unix_users.add_plugin(UnixLocalePlugin)
 
     assert target_unix_users.language == ["en_US"]
+
+
+def test_locale_etc_localtime_symlink(target_unix_users, fs_unix):
```
Comment (Schamper) [inline tests/test_plugins_os_unix_locale.py:36]: Can you add type hints to the tests?

**Review by Schamper — CHANGES_REQUESTED** [review]

File: tests/test_plugins_os_unix_locale.py (hunk 2 above)
Comment (yunzheng) [reply tests/test_plugins_os_unix_locale.py]: done

File: tests/test_plugins_os_unix_locale.py [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
+    VirtualSymlink,
+)
 from dissect.target.plugins.os.unix.locale import LocalePlugin as UnixLocalePlugin
+from dissect.target.target import Target
 
 from ._utils import absolute_path
 
 
-def test_locale_plugin_unix(target_unix_users, fs_unix):
+def test_locale_plugin_unix(target_unix_users: Target, fs_unix: VirtualFilesystem):
```
Comment (Schamper) [inline tests/test_plugins_os_unix_locale.py:16]:
```suggestion
def test_locale_plugin_unix(target_unix_users: Target, fs_unix: VirtualFilesystem) -> None:
```

File: tests/test_plugins_os_unix_locale.py [hunk 4]
```diff
@@ -25,23 +31,23 @@ def test_locale_plugin_unix(target_unix_users, fs_unix):
     assert keyboard[0].backspace == "guess"
 
 
-def test_locale_plugin_unix_quotes(target_unix_users, fs_unix):
+def test_locale_plugin_unix_quotes(target_unix_users: Target, fs_unix: VirtualFilesystem):
```
Comment (Schamper) [inline tests/test_plugins_os_unix_locale.py:34]:
```suggestion
def test_locale_plugin_unix_quotes(target_unix_users: Target, fs_unix: VirtualFilesystem) -> None:
```

File: tests/test_plugins_os_unix_locale.py [hunk 5]
```diff
@@ -25,23 +31,23 @@ def test_locale_plugin_unix(target_unix_users, fs_unix):
...
     # Older Fedora system
     fs_unix.map_file_fh("/etc/default/locale", BytesIO(b'LANG="en_US.UTF-8"'))
     target_unix_users.add_plugin(UnixLocalePlugin)
 
     assert target_unix_users.language == ["en_US"]
 
 
-def test_locale_etc_localtime_symlink(target_unix_users, fs_unix):
+def test_locale_etc_localtime_symlink(target_unix_users: Target, fs_unix: VirtualFilesystem):
```
Comment (Schamper) [inline tests/test_plugins_os_unix_locale.py:42]:
```suggestion
def test_locale_etc_localtime_symlink(target_unix_users: Target, fs_unix: VirtualFilesystem) -> None:
```

File: tests/test_plugins_os_unix_locale.py [hunk 6]
```diff
@@ -25,23 +31,23 @@ def test_locale_plugin_unix(target_unix_users, fs_unix):
...
     fs_unix.symlink("/usr/share/zoneinfo/Europe/Amsterdam", "/etc/localtime")
     target_unix_users.add_plugin(UnixLocalePlugin)
 
     assert isinstance(fs_unix.get("/etc/localtime"), VirtualSymlink)
     assert target_unix_users.timezone == "Europe/Amsterdam"
 
 
-def test_locale_etc_localtime_hardlink(target_unix_users, fs_unix):
+def test_locale_etc_localtime_hardlink(target_unix_users: Target, fs_unix: VirtualFilesystem):
```
Comment (Schamper) [inline tests/test_plugins_os_unix_locale.py:50]:
```suggestion
def test_locale_etc_localtime_hardlink(target_unix_users: Target, fs_unix: VirtualFilesystem) -> None:
```

File: tests/test_plugins_os_unix_locale.py [hunk 7]
```diff
@@ -58,7 +64,7 @@ def test_locale_etc_localtime_hardlink(target_unix_users, fs_unix):
         assert target_unix_users.timezone == "Europe/Amsterdam"
 
 
-def test_locale_etc_localtime_regular_file(target_unix_users, fs_unix):
+def test_locale_etc_localtime_regular_file(target_unix_users: Target, fs_unix: VirtualFilesystem):
```
Comment (Schamper) [inline tests/test_plugins_os_unix_locale.py:67]:
```suggestion
def test_locale_etc_localtime_regular_file(target_unix_users: Target, fs_unix: VirtualFilesystem) -> None:
```

**Review by Schamper — CHANGES_REQUESTED** [review]

**Review by Schamper — APPROVED** [review]

**Pull request merged**